### PR TITLE
api joined-offers tests

### PR DESCRIPTION
**Pull request by kristhina**

### CODE REVIEW

File: backend/apps/volontulo/tests/views/api/test_joined_offers.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+from django.urls import reverse
+from rest_framework.test import APITestCase
+
+from apps.volontulo.factories import UserFactory
+from apps.volontulo.factories import OfferFactory
+
+ENDPOINT_URL = reverse('joined_offers')
+
+
+class TestJoinedOffers(APITestCase, TestCase):
```
[the remark this answers is not in the document]
Comment (kristhina) [reply backend/apps/volontulo/tests/views/api/test_joined_offers.py]: changed

File: backend/apps/volontulo/tests/views/api/test_joined_offers.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        # tests the length of the list of offers that user joined
+        self.assertEqual(res.data, [])
+
+    def test_user_joined_one_offer(self):
+        """Tests if user joined one offer"""
+
+        user = UserFactory()
+        self.client.force_login(user=user)
+
+        offer = OfferFactory(image=None)
```
[the remark this answers is not in the document]
Comment (kristhina) [reply backend/apps/volontulo/tests/views/api/test_joined_offers.py]: I had some problems with it earlier but now everything worked ok so I removed it

File: backend/apps/volontulo/tests/views/api/test_joined_offers.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        offer.volunteers.add(user)
+
+        res = self.client.get(ENDPOINT_URL)
+
+        res_id = res.data[0]['id']
+        offer_id = offer.id
```
[the remark this answers is not in the document]
Comment (kristhina) [reply backend/apps/volontulo/tests/views/api/test_joined_offers.py]: changed

File: backend/apps/volontulo/tests/views/api/test_joined_offers.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+        res_title = res.data[0]['title']
```
[the remark this answers is not in the document]
Comment (kristhina) [reply backend/apps/volontulo/tests/views/api/test_joined_offers.py]: changed

File: backend/apps/volontulo/tests/views/api/test_joined_offers.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        self.assertEqual(res_id, offer_id)
+
+        # tests the title of the offer created and the offer that user joined
+        self.assertEqual(res_title, offer_title)
+
+    def test_user_joined_some_offers(self):
+        """Tests if user joined more than one offer"""
+
+        user = UserFactory()
+        user1 = UserFactory()
```
[the remark this answers is not in the document]
Comment (kristhina) [reply backend/apps/volontulo/tests/views/api/test_joined_offers.py]: I wanted to check if it displays only offers that active user joined (not offers joined by other volunteers). I needed another user to check it.

File: docker-compose-no-backend.yml [hunk 6]
```diff
@@ -0,0 +1,19 @@
+frontend:
```
[the remark this answers is not in the document]
Comment (kristhina) [reply docker-compose-no-backend.yml]: it helps me debugging backend locally

File: backend/apps/volontulo/tests/views/api/test_joined_offers.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (magul) [reply backend/apps/volontulo/tests/views/api/test_joined_offers.py]: So maybe instead
```
user2 = UserFactory()
...
offer3 = OfferFactory()
offer3.volunteers.add(user2)
```
You can do something like this
```
OfferFactory(volunteers=UserFactory.create_batch(10))
```

**Pull request merged**